utils/ibmsetup.py

```diff
@@ -0,0 +1,167 @@
+import ipaddress
+import json
+import os
+import re
+import socket
+import subprocess
+import sys
+
+import ibmconf
+
+def ibmcloud(cmd):
+    cmd = ["ibmcloud"]+cmd+["--output", "JSON"]
+    try:
+        data = subprocess.run(cmd, capture_output=True, text=True, check=True).stdout
+    except subprocess.CalledProcessError as E:
+        exit("ERROR: %s\n%s"%(" ".join(cmd), E.stderr))
+    return json.loads(data)
+
+def ppjson(d):
+    print(json.dumps(d, indent=2))
+
+class Host(object):
+    def __init__(self, detail):
+        self.data = {}
+        self.data["hostname"] = detail["hostname"]
+        self.data["id"] = detail["id"]
+
+        self.data["vlans"] = {}
+        for vlan in detail["networkVlans"]:
+            self.data["vlans"][vlan["networkSpace"]] = vlan["vlanNumber"]
+            self.data["vlans"][vlan["networkSpace"]+"Id"] = vlan["id"]
+
+        for net in detail["networkComponents"]:
+            if net.get("ipmiIpAddress"):
+                self.data["ipmi_ip"] = net["ipmiIpAddress"]
+                self.data["ipmi_user"] = detail["remoteManagementAccounts"][0]["username"]
+                self.data["ipmi_password"] = detail["remoteManagementAccounts"][0]["password"]
+            elif net.get("primaryIpAddress") == detail["primaryIpAddress"]:
+                self.data["publicip"] = net["primaryIpAddress"]
+                self.data["publicmac"] = net["macAddress"]
+            elif net.get("primaryIpAddress") == detail["primaryBackendIpAddress"]:
+                self.data["privatemac"] = net["macAddress"]
+                self.data["privateip"] = net["primaryIpAddress"]
+
+    def __repr__(self):
+        return "%s - ipmi:%s, prov_mac:%s"%(self.data["hostname"], self.data["ipmi_ip"], self.data["privatemac"])
+
```

Comment:
it may be worth to split this script in two parts, one for the class Host (maybe IbmHost ?) and eventually functions, and one with the actual script

Reply:
ya, if it looks like its gonna get more complicated a split could be worthwhile